lib/bankaccount/savingsaccount.rb

```diff
@@ -0,0 +1,29 @@
+module BankAccount
+  class SavingsAccount < Account
+
+    def initialize(id, initial_balance, required_opening_balance = 10) # [breakfast]
+      # the initial balance cannot be less than $10. If it is, this will raise an ArgumentError [breakfast]
+      super
+    end
+
+    ##QUESTION: Is this way preferred, or is the commented out version below preferred?
+    def withdraw(amount, transaction_fee = 2, min_balance = 10) # [breakfast]
+      # each withdrawal incurs a $2 fee (taken out of balance) [breakfast]
+      super
+    end
+
+    # def withdraw(amount) # [breakfast]
+    #   # each withdrawal incurs a $2 fee (taken out of balance) [breakfast]
+    #   super(amount, 2, 10) # (amount, transaction_fee, min_balance)
+    # end
+
```

Comment:
Completely makes sense to me to not un-necessarily expose the variables via the params.

To an extent, if I wanted to take advantage of the inheritance, I still have to include "extra" things in the parent method though, right? Since Account doesn't actually have transaction fees or a minimum balance, it's just the children accounts that do?